iceoryx_utils/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -71,10 +73,35 @@ class UnixDomainSocket_test : public Test
 const size_t UnixDomainSocket_test::MaxMsgSize = UnixDomainSocket::MAX_MESSAGE_SIZE;
 constexpr uint64_t UnixDomainSocket_test::MaxMsgNumber;
 
+TEST_F(UnixDomainSocket_test, NonBlockingModeNotSupported)
+{
+    auto result = UnixDomainSocket::create(
+        goodName, IpcChannelMode::NON_BLOCKING, IpcChannelSide::SERVER, MaxMsgSize, MaxMsgNumber);
+    EXPECT_TRUE(result.has_error());
+    ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INVALID_ARGUMENTS));
+}
+
 TEST_F(UnixDomainSocket_test, UnlinkNonExistingWithInvalidNameLeadsToError)
 {
     auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, invalidName);
     EXPECT_TRUE(ret.has_error());
     ASSERT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
 }
+
+TEST_F(UnixDomainSocket_test, SendingOnServerLeadsToError)
```

Comment:
Somehow I think client and server are not the best names for this. I understand what's going on and that two channels are needed for bidirectional communication, but it's weird to read that a server cannot send something. Maybe `RequestChannel` and `ResponseChannel` would fit better. But that would be something for a different PR.

Reply:
@elfenpiff ^